pkg/controller/ecs/service/setup.go

```diff
@@ -533,11 +536,11 @@ func generateLoadBalancers(cr *svcapitypes.Service) []*svcsdk.LoadBalancer {
 
 // Helper func to generate SDK NetworkConfiguration types from the crossplane
 // types included in the Service api.
-func generateNetworkConfiguration(cr *svcapitypes.Service) *svcsdk.NetworkConfiguration {
+func generateNetworkConfiguration(cr *svcapitypes.Service) (*svcsdk.NetworkConfiguration, error) {
 	networkConfiguration := &svcsdk.NetworkConfiguration{}
 
 	if cr.Spec.ForProvider.NetworkConfiguration == nil {
-		return networkConfiguration
+		return nil, errors.New("network configuration is nil")
```

Comment:
We should not misuse an error as an indicator here if we just want to not set a value.

Instead just return nil in this case so `obj.NetworkConfiguration` is just nil.